test/parallel/test-tls-env-extra-ca-with-ca.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const common = require('../common');
+
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+const assert = require('assert');
+const tls = require('tls');
+const fixtures = require('../common/fixtures');
+
+const { fork } = require('child_process');
+
+if (process.env.CHILD) {
+  const copts = {
+    port: process.env.PORT,
+    checkServerIdentity: common.mustCall()
+  };
+
+  // New secure contexts have the well-known root CAs.
+  copts.secureContext = tls.createSecureContext();
+
+  // Explicit calls to addCACert() add to the root certificates,
+  // instead of replacing, so connection still succeeds.
+  copts.secureContext.context.addCACert(
+    fixtures.readKey('ca1-cert.pem')
+  );
+
+  const client = tls.connect(copts, common.mustCall(() => {
+    client.end('hi');
+  }));
+
+  return;
+}
+
+const options = {
+  key: fixtures.readKey('agent3-key.pem'),
+  cert: fixtures.readKey('agent3-cert.pem')
```

Comment:
The structure of these is non-intuitive but was (somewhat) intentional.

**ca-with-pfx**:
* There are three 'agent' PFX files, `agent1.pfx`, `agent6.pfx`, and `agent10.pfx`. For the test, I selected the lowest numbered of them (`agent1`). 
* Since `agent1.pfx` is signed by `ca1`, we can't use `ca1` as part of `NODE_EXTRA_CA_CERTS` or a server certificate chain since it would invalidate the test. 
* `agent2-cert.pem` is self-signed, so we can't use that for the server certificate.
* `agent3-cert.pem` is signed by `ca2` and `ca2` does not have a trust relationship back to `ca1` - that makes it the 'first' appropriate server certificate that we can use.
* Since `agent3-cert.pem` is signed by `ca2`, I use `ca2-cert.pem` as `NODE_EXTRA_CA_CERTS`.

It looks almost random having the PFX, server certificate, and EXTRA_CA_CERTS all different numbers, but it's the first option that works.

**ca-with-crl**:
* There is only one CRL, `ca2-crl.pem`.
* `ca2-crl` is associated with the `ca2` certificate authority. 
* `agent3` is the first key/certificate pair (e.g. server key) associated with `ca2`, so it's used for the server.
* `agent3` is signed by `ca2`, so `ca2-cert.pem` is use as `NODE_EXTRA_CA_CERTS

**ca-with-ca**:
* I need two certificate authorities - one for `NODE_EXTRA_CA_CERTS` and one for the 'additional' cert added to the context. I use `ca1` and `ca2` for that purpose, with `ca1` being the "additional one".
* Since `ca1` is used as the additional cert, `ca2` is used as `NODE_EXTRA_CA_CERTS`
* `agent3` is the first server certificate/key signed under `ca2`, so it's used for the server.

Short version is that it looks almost random due to the structure of the filenames, but they were selected based on a set of criteria. There are many other options, of course!